pkg/cli/admin/release/extract.go

```diff
@@ -196,23 +210,77 @@ func (o *ExtractOptions) Run() error {
 				To:   dir,
 			},
 		}
+		var manifestErrs []error
 		found := false
 		opts.TarEntryCallback = func(hdr *tar.Header, _ extract.LayerInfo, r io.Reader) (bool, error) {
-			if hdr.Name != o.File {
+			if !o.ExtractManifests {
+				if hdr.Name != o.File {
+					return true, nil
+				}
+				if _, err := io.Copy(o.Out, r); err != nil {
+					return false, err
+				}
+				found = true
+				return false, nil
+			} else {
+				switch hdr.Name {
+				case o.File:
+					if _, err := io.Copy(o.Out, r); err != nil {
+						return false, err
+					}
+					found = true
+				case "image-references":
+					return true, nil
+				case "release-metadata":
+					return true, nil
+				default:
+					if ext := path.Ext(hdr.Name); len(ext) > 0 && (ext == ".yaml" || ext == ".yml" || ext == ".json") {
+						klog.V(4).Infof("Found manifest %s", hdr.Name)
+						raw, err := ioutil.ReadAll(r)
+						if err != nil {
+							manifestErrs = append(manifestErrs, errors.Wrapf(err, "error reading file %s", hdr.Name))
```

Comment:
I go through all the manifests and gather any errors encountered. I only care about the two signature manifest files and I've noticed in testing that some of the other manifest files generate errors. After going through all manifests errors will be output if the signature manifests were not found since one or more of the errors will most likely apply. If the signature manifests are found I do not print out the errors since they should be irrelivant and will certainly cause confusion and alarm to the user.

Reply:
Right, but you don't have any kind of loop in your code, that's handled where `TarEntryCallback` is being used. Thus you need to stop the execution when you encounter an error, if you don't your code will start panicing trying to read empty `raw`.

Reply:
Sorry, this shouldn't have been so hard, had tunnel vision. Returns are in...

Reply:
Inserted return statement